### ISSUE TITLE
Build fails on Fedora 27 and 28

### ISSUE THREAD

**Issue by giostro** [cisco/ChezScheme#301]
.....
prim5.c: In function ‘s_showalloc’:
prim5.c:329:85: error: ‘td of ’ directive output may be truncated writing 6 bytes into a region of size between 0 and 18 [-Werror=format-truncation=]
   snprintf(fmtbuf, FMTBUFSIZE, "%%#0%dtx %%#0%dtx (+ %%#0%dtx bytes @ %%#0%dtx) %%%dtd of %%%dtd\n",
                                                                                     ^~~~~~
prim5.c:329:3: note: ‘snprintf’ output between 54 and 112 bytes into a destination of size 60
   snprintf(fmtbuf, FMTBUFSIZE, "%%#0%dtx %%#0%dtx (+ %%#0%dtx bytes @ %%#0%dtx) %%%dtd of %%%dtd\n",
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       addrwidth, byteswidth, headerbyteswidth, headeraddrwidth, segswidth, segswidth);
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:29: prim5.o] Error 1
make[1]: *** [Makefile:20: build] Error 2
make: *** [Makefile:22: install] Error 2


I can't find any other similar errors online. I'm not knowledgeable enough to determine whether this is an issue with s_showalloc in prim5.c or GCC's snprintf. 

G

**Comment by giostro** [comment]
Thank you.
9.5 still won't build after patching because it can't find xlocale.h. However, master builds fine.

**Comment by lijinpei** [comment]
make CFLAGS=-Wno-format-truncation can temporarily disable the warning, but not the final solution.

**Comment by akeep** [comment]
I've merged #299, which as @jltaylor-us notes addresses the problem with gcc-8.  Thanks to @weinholt for the patch.

Source pulled from GitHub should now build with gcc-8, and I've successfully tested on Fedora 28.

**Comment by amirouche** [comment]
Maybe one can close the issue if it works. :100:
